crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -513,6 +513,27 @@ impl SlidingSync {
         if to_device_enabled {
             request.extensions.to_device.since =
                 restored_fields.and_then(|fields| fields.to_device_token);
+
+            // There is a non-negligible difference MSC3575 and MSC4186 in how
+            // the `e2ee` extension works. When the client sends a request with
+            // no `since` token:
+            //
+            // * MSC3575 returns all device lists updates since the last
+            //   request from the device that asked for device lists (this works
+            //   similarly to to-device message handling),
+            // * MSC4186 returns no device lists updates, as it only returns
+            //   changes since the provided `since` token (which is `null` in
+            //   this case); this is in line with sync v2.
+            //
+            // Therefore, with MSC4186, the device list cache must be reset if
```

Comment:
We're not really clearing the cache, we're marking it that it needs to be re-downloaded. Not sure if you'd like to clarify that a bit.

Reply:
Indeed. Gonna modify.